lib/activerecord-import/adapters/mysql_adapter.rb

```diff
@@ -71,14 +71,12 @@ def pre_sql_statements( options)
 
   # Add a column to be updated on duplicate key update
   def add_column_for_on_duplicate_key_update( column, options = {} ) # :nodoc:
-    if options.include?(:on_duplicate_key_update)
+    if options.include?(:on_duplicate_key_update) && !options[:ignore] && !options[:on_duplicate_key_ignore]
```

Comment:
With this change, I think we could simplify this to:

`if options[:on_duplicate_key_update] && !options[:ignore] && !options[:on_duplicate_key_ignore]`